components/camera/rtsp/rtsp_test.go

```diff
@@ -22,34 +26,55 @@ func startFFMPEG(outputURL string, streamStarted chan struct{}) context.CancelFu
 		ffmpegStream = ffmpegStream.Output(outputURL, ffmpeg.KwArgs{"vcodec": "mjpeg", "huffman": "0", "f": "rtsp", "rtsp_transport": "tcp"})
 		ffmpegStream.Context = cancelCtx
 		cmd := ffmpegStream.OverWriteOutput().Compile()
-		close(streamStarted)
 		if err := cmd.Run(); err != nil {
 			return
 		}
 	})
 	return cancel
 }
 
-func TestRTSPCamera(t *testing.T) {
-	logger := golog.NewTestLogger(t)
-	// set up the rtsp simple server running on rtsp://localhost:8598/mystream
-	outputURL := "rtsp://127.0.0.1:8598/mystream"
+func startRTSPServer(t *testing.T) func() {
+	t.Helper()
 	configLoc := []string{artifact.MustPath("components/camera/rtsp/rtsp-simple-server.yml")}
 	s, ok := server.New(configLoc)
 	test.That(t, ok, test.ShouldBeTrue)
-	defer s.Close()
-	streamStarted := make(chan struct{})
-	cancel := startFFMPEG(outputURL, streamStarted)
-	defer cancel()
-	// create the rtsp camera model
+	return s.Close
+}
+
+func TestRTSPCamera(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	host := "127.0.0.1"
+	port := "32512"
```

Comment:
It can be anything - but just wanted to make sure it was large and unlikely to clash with a real port. The only thing is that there is a yaml file for the server that needs to have the same port number for this to work.